sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs

```diff
@@ -2306,6 +2306,86 @@ async Task ProcessMessage(ProcessSessionMessageEventArgs args)
+            {
+                await using var client = CreateClient();
+                List<ServiceBusSessionProcessor> processors = new();
+                int numProcessors = 50;
+                for (int i = 0; i < numProcessors; i++)
+                {
+                    ServiceBusSessionProcessorOptions options = new()
+                    {
+                        AutoCompleteMessages = true,
+                        MaxConcurrentSessions = 1,
+                        MaxConcurrentCallsPerSession = 100,
+                        PrefetchCount = 0,
+                        SessionIdleTimeout = TimeSpan.FromSeconds(4),
+                        MaxAutoLockRenewalDuration = TimeSpan.FromMinutes(5)
+                    };
+                    ServiceBusSessionProcessor processor = client.CreateSessionProcessor(scope.QueueName, options);
+                    processor.ProcessMessageAsync += ProcessMessageAsync;
+                    processor.ProcessErrorAsync += SessionErrorHandler;
+                    await processor.StartProcessingAsync();
+                    processors.Add(processor);
+                }
+
+                var sender = client.CreateSender(scope.QueueName);
+                CancellationTokenSource cts = new();
+                cts.CancelAfter(TimeSpan.FromMinutes(4));
+                await SendMessagesAsync(cts.Token);
+                foreach (var processor in processors)
+                {
+                    await processor.CloseAsync();
+                }
+
+                async Task SendMessagesAsync(CancellationToken cancellationToken)
+                {
+                    while (!cancellationToken.IsCancellationRequested)
+                    {
+                        try
+                        {
+                            await Task.Delay(10000, cancellationToken);
+                        }
+                        catch (OperationCanceledException)
+                        {
+                            return;
+                        }
+
+                        Random rnd = new Random();
+                        int numSessions = rnd.Next(0, 30);
+                        var messages = new List<ServiceBusMessage>();
+                        for (int i = 0; i < numSessions; i++)
+                        {
+                            var sessionId = Guid.NewGuid().ToString();
+                            messages.Add(new ServiceBusMessage("Hello World") { SessionId = sessionId });
+                            if (i % 2 == 0)
+                            {
+                                messages.Add(new ServiceBusMessage("Hello World") { SessionId = sessionId });
+                            }
+                        }
+
+                        await sender.ScheduleMessagesAsync(messages, DateTimeOffset.Now.AddSeconds(10));
+                    }
+                }
+
+                Task ProcessMessageAsync(ProcessSessionMessageEventArgs args)
```

Comment:
Should we assert that we've been called at least once and, ideally, the expected number of times?